crates/notion-core/src/tool/mod.rs

```diff
@@ -69,6 +72,28 @@ impl ToolSpec {
         }
         Ok(())
     }
+
+    pub fn try_from_str(tool_spec: &str) -> Fallible<Self> {
+        let re = Regex::new("(?P<name>@?[-/_A-z]+)(@(?P<version>.+))?").expect("regex is valid");
+        let captures = re
+            .captures(tool_spec)
+            .ok_or(ErrorDetails::ParseToolSpecError {
+                tool_spec: tool_spec.into(),
+            })?;
+
+        let version = captures
+            .name("version")
+            .map(|version| VersionSpec::parse(version.as_str()))
+            .transpose()?
+            .unwrap_or_else(VersionSpec::default);
+
+        Ok(match &captures["name"] {
+            "node" => ToolSpec::Node(version),
+            "yarn" => ToolSpec::Yarn(version),
```

Comment:
Change this to match the new ordering in `ToolSpec` above?

Reply:
Great suggestion; I'm doing likewise throughout the module.